support-frontend/assets/components/paymentFrequencyTabs/paymentFrequencyTabsContainer.tsx

```diff
@@ -34,6 +36,14 @@ export function PaymentFrequencyTabsContainer({
 	);
 	const productType = useContributionsSelector(getContributionType);
 
+	// Force to monthly if hideOneOff, OneOff and refresh pressed
+	if (
+		hideOneOff &&
+		useContributionsSelector(getContributionType) === 'ONE_OFF'
+	) {
```

Comment:
```suggestion
 const contributionType = useContributionsSelector(getContributionType);
 
	if (
		hideOneOff &&
		contributionType === 'ONE_OFF'
	) {
```
The right-hand side of a logical operator in JS (ie. `&&`, `||`) is lazily evaluated- this means that if the left hand side is falsey for `&&` or truthy for `||`, any code on the right hand side will not be run. Try running this in your console to see:

```javascript
false && console.log("you won't see this");
true || console.log('or this');
```

Because of the way React hooks work it's really important that they get called _every single time_ a component renders, so that React can associate the correct bits of state with the correct components. We should always use hooks at the top level, and then use their results in conditions if we need to.

Reply:
Ok, don't put React  hook call into a conditional where it may not be called. Change applied.